File: src/main/java/com/toocol/ssh/core/file/core/DirectoryChooser.java
package com.toocol.ssh.core.file.core;

import com.toocol.ssh.common.jni.TermioJNI;

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/4/15 22:07
 * @version: 0.0.1
 */
public class DirectoryChooser {

    private static final TermioJNI TERMIO_JNI = TermioJNI.getInstance();

    /**
     * chose the local path
     *
     * @return file paths
     */
    public String showOpenDialog() {
        return TERMIO_JNI.chooseDirectory();
    }

}

File: core/src/main/java/com/toocol/termio/core/file/core/DirectoryChooser.kt
package com.toocol.termio.core.file.core

import com.toocol.termio.utilities.console.Console

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/4/15 22:07
 * @version: 0.0.1
 */
class DirectoryChooser {
    /**
     * chose the local path
     *
     * @return file paths
     */
    fun showOpenDialog(): String? {
        return console.chooseDirectory()
    }

    companion object {
        private val console = Console.get()
    }
}